src/propmachinery.jl

```diff
@@ -1,32 +1,45 @@
-import Base: start, next, done
-
 abstract QuPropagatorMethod
 
-immutable QuPropagator{QPM<:QuPropagatorMethod}
-    hamiltonian
-    init_state::QuVector
+immutable QuPropagator{QPM<:QuPropagatorMethod, QVM<:Union(QuBase.AbstractQuVector,QuBase.AbstractQuMatrix)}
+    eq
```

Comment:
@acroy I have tried to address this but this would allow cases like 
`QuSchrodingerEq` and `AbstractQuMatrix`, `QuLiouvillevonNeumannEq` and `AbstractQuVector` both of which can be brought to the constructs already present by constructing the `superoperator` from the `hamiltonian` of the `QuSchrodinderEq` in the former and converting the `AbstractQuVector` to `AbstractQuMatrix` in the latter. Any thoughts on this would be helpful.